lib/shared/src/prompt/truncation.ts

```diff
@@ -1,13 +1,15 @@
 import type { RangeData } from '../common/range'
-
-import { CHARS_PER_TOKEN } from './constants'
+import { TokenCounter } from '../token'
 
 /**
  * Truncates text to the given number of tokens, keeping the start of the text.
  */
 export function truncateText(text: string, maxTokens: number): string {
-    const maxLength = maxTokens * CHARS_PER_TOKEN
-    return text.length <= maxLength ? text : text.slice(0, maxLength)
+    const encoded = TokenCounter.encode(text)
+    if (encoded.length < maxTokens) {
```

Comment:
This could me more succinct and save some compute sometimes with `<=`
```suggestion
    return encoded.length <= maxTokens ? text : TokenCounter.decode(encoded.slice(0, maxTokens))
```

Reply:
Oh wow i didn't know that, thank you for the tips!